### PR TITLE
docs(config): add note about GnuPG v2.4 usage

### PR DESCRIPTION
**Pull request by viceice**
## Changes
Add a note to disable the `AEAD` GnuPG v2.4 extensions.

## Context
- https://github.com/renovatebot/renovate/discussions/28934#discussioncomment-9419010

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -929,6 +929,31 @@ uid                      Renovate Bot <renovate@whitesourcesoftware.com>
...
+     Features: MDC, AEAD, Keyserver no-modify
+
+gpg> setpref AES256 AES192 AES 3DES SHA512 SHA384 SHA256 SHA224 SHA1 ZLIB BZIP2 ZIP
+Set preference list to:
+     Cipher: AES256, AES192, AES, 3DES
+     AEAD:
+     Digest: SHA512, SHA384, SHA256, SHA224, SHA1
+     Compression: ZLIB, BZIP2, ZIP, Uncompressed
+     Features: MDC, Keyserver no-modify
+Really update the preferences? (y/N) y
```
Comment (viceice) [inline docs/usage/self-hosted-configuration.md:954]:
```suggestion
Really update the preferences? (y/N) y
gpg> save
```

Not sure if the save is required. maybe better add this to be explicit.

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -929,6 +929,32 @@ uid                      Renovate Bot <renovate@whitesourcesoftware.com>
 sub   rsa4096 2021-09-10 [E]
 ```
 
+<!-- prettier-ignore -->
+!!! note
+    If you use gnupg v2.4 or newer to generate the key, then you need to disable `AEAD` preferences.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:934]:
```suggestion
    If you use GnuPG `v2.4` (or newer) to generate the key, then you must disable `AEAD` preferences.
```

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -929,6 +929,32 @@ uid                      Renovate Bot <renovate@whitesourcesoftware.com>
...
+    Otherwise Renovate can't decrypt the encrypted values.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:935]:
```suggestion
    This is needed to allow  Renovate to decrypt the encrypted values.
```

File: docs/usage/self-hosted-configuration.md [hunk 4]
```diff
@@ -929,6 +929,32 @@ uid                      Renovate Bot <renovate@whitesourcesoftware.com>
...
+
+```
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:937]:
```suggestion
```bash
```

Maybe CLI styling the code looks nicer?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**